app/assets/javascripts/oxalis/model/sagas/prefetch_saga.js

```diff
@@ -0,0 +1,137 @@
+// @flow
+import Model from "oxalis/model";
+import constants from "oxalis/constants";
+import { throttle, select, take, call } from "redux-saga/effects";
+import {
+  getPosition,
+  getRequestLogZoomStep,
+  getAreas,
+} from "oxalis/model/accessors/flycam_accessor";
+import {
+  PrefetchStrategySkeleton,
+  PrefetchStrategyVolume,
+} from "oxalis/model/bucket_data_handling/prefetch_strategy_plane";
+import { PrefetchStrategyArbitrary } from "oxalis/model/bucket_data_handling/prefetch_strategy_arbitrary";
+import { FlycamActions } from "oxalis/model/actions/flycam_actions";
+import DataLayer from "oxalis/model/data_layer";
+
+const PREFETCH_THROTTLE_TIME = 50;
+const DIRECTION_VECTOR_SMOOTHER = 0.125;
+
+const prefetchStrategiesArbitrary = [new PrefetchStrategyArbitrary()];
+const prefetchStrategiesPlane = [new PrefetchStrategySkeleton(), new PrefetchStrategyVolume()];
+
+export function* watchDataRelevantChanges(): Generator<*, *, *> {
+  yield take("WK_READY");
+
+  const previousProperties = { lastDirection: [0, 0, 0] };
+  yield throttle(PREFETCH_THROTTLE_TIME, FlycamActions, triggerDataPrefetching, previousProperties);
+}
+
+function* shouldPrefetchForDataLayer(dataLayer: DataLayer): Generator<*, *, *> {
+  const isSegmentationLayer = dataLayer.category === "segmentation";
+  const segmentationOpacity = yield select(state => state.datasetConfiguration.segmentationOpacity);
+  const isSegmentationVisible = segmentationOpacity !== 0;
+  // There is no need to prefetch data for segmentation layers that are not visible
+  return !isSegmentationLayer || isSegmentationVisible;
+}
+
+export function* triggerDataPrefetching(previousProperties: Object): Generator<*, *, *> {
+  const currentViewMode = yield select(state => state.temporaryConfiguration.viewMode);
+  const isPlaneMode = constants.MODES_PLANE.includes(currentViewMode);
+
+  if (isPlaneMode) {
+    const dataLayers = yield call([Model, Model.getAllLayers]);
+    for (const dataLayer of dataLayers) {
+      if (yield call(shouldPrefetchForDataLayer, dataLayer)) {
+        yield call(prefetchForPlaneMode, dataLayer, previousProperties);
+      }
+    }
+  } else {
+    const dataLayers = yield call([Model, Model.getColorLayers]);
+    for (const colorLayer of dataLayers) {
+      yield call(prefetchForArbitraryMode, colorLayer, previousProperties);
+    }
+  }
+}
+
+export function* prefetchForPlaneMode(
+  layer: DataLayer,
+  previousProperties: Object,
+): Generator<*, *, *> {
+  const position = yield select(state => getPosition(state.flycam));
+  const zoomStep = yield select(state => getRequestLogZoomStep(state));
+  const activePlane = yield select(state => state.viewModeData.plane.activeViewport);
+  const tracingType = yield select(state => state.tracing.type);
+  const { lastPosition, lastDirection, lastZoomStep } = previousProperties;
+
+  let direction = lastDirection;
```

Comment:
I'd change this to `let direction = lastDirection || [0, 0, 0];` so that you can change the default of `previousProperties` to `{}` in line 27.